README.md

```diff
@@ -1 +1,182 @@
+
+These steps are required because the widget needs a valid robot and Viam instance to communicate with the Viam cloud and provision the robot.
+
+### Basic Example
+
+```dart
+import 'package:viam_flutter_hotspot_provisioning_widget/viam_flutter_hotspot_provisioning_widget.dart';
+
+// 1. Initialize a Viam instance with your API credentials
+final viam = await Viam.withApiKey(apiKeyId, apiKey);
+
+// 2. Create a new robot or get an existing one
+final robot = await viam.appClient.getRobot(robotId);
+// OR create a new robot:
+// final robotId = await viam.appClient.newMachine(robotName, locationId);
+// final robot = await viam.appClient.getRobot(robotId);
+
+// 3. Get the main robot part
+final mainPart = (await viam.appClient.listRobotParts(robot.id))
+    .firstWhere((element) => element.mainPart);
+
+// 4. Start the provisioning flow
+final result = await HotspotProvisioningFlow.show(
+  context,
+  robot: robot,
+  viam: viam,
+  mainPart: mainPart,
+  hotspotPrefix: 'your-hotspot-prefix',  // Must match viam-defaults.json
+  hotspotPassword: 'your-hotspot-password', // Must match viam-defaults.json
+);
+
+// 5. Handle the result
+if (result != null) {
+  if (result.status == RobotStatus.online) {
+    // Robot successfully provisioned. Robot is online
+    print('Robot ${result.robot.name} is online!');
+  } else {
+    // Provisioning failed or timed out. Robot is offline
+    print('Robot provisioning failed');
+  }
+}
+```
+## HotspotProvisioningFlow Widget
+The main widget that handles the entire provisioning flow.
+
+### Constructor Parameters:
+What you need to pass into the widget:
+- `robot`: The Viam robot to provision
+- `viam`: The Viam SDK instance
+- `mainPart`: The main robot part
+- `hotspotPrefix`: The SSID prefix for the robot's hotspot. This prefix **must match** the prefix you set in the viam-defaults.json.
+- `hotspotPassword`: The password for the robot's hotspot. This password **must match** the password you set in the viam-defaults.json.
+
+### HotspotProvisioningResult
+Contains the result of the provisioning attempt:
+- `robot`: The robot that was provisioned
+- `status`: The robot's status (online/offline)
+
+### Additional Features
+- **Manual Network Entry**: Fallback option to manually enter network credentials when automatic detection fails
+- **Error Handling**: User-friendly error messages for common issues like incorrect hotspot password.
+- **Network Type Indicators**: Icons to distinguish between public and private Wi-Fi networks
+
+### Dependencies
+This package depends on:
+- `plugin_wifi_connect`: For Wi-Fi connection functionality
+- `viam_sdk`: For Viam robot communication
+- `permission_handler`: For platform permissions
+- `flutter_platform_widgets`: For platform-specific UI
+- `provider`: For state management
+
+### Looking for a complete example?
+See the `example/` directory for a complete working example app that allows you to provision Viam devices.
```

Comment:
nit you can link directly to that directory so it's easy to click into that from this file [example](example) may do it